utils/SolanaNextAuth.ts

```diff
@@ -0,0 +1,30 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import SolanaAuth, { SolanaAuthOptions } from "./SolanaAuth";
+
+/**
+ * This function should be used on a catch-all Next API route
+ * It's a wrapper for the SolanaAuth Constructor
+ * /api/auth/[...solanaAuth].ts
+ */
+const SolanaNextAuth = (options: SolanaAuthOptions) => {
+  let handler = SolanaAuth(options);
+
+  return async (req: NextApiRequest, res: NextApiResponse) => {
+    const { solana } = req.query;
+
+    switch (solana[0]) {
```

Comment:
Be sure to update this if you end up changing the route as per my comment from above